robot/src/main/java/us/ilite/robot/driverinput/DriverInput.java

```diff
@@ -61,87 +68,108 @@ public void periodicInput(double pNow) {
+        /*
+        If we aren't already running commands and the driver is pressing a button that triggers a command,
+        set the superstructure command queue based off of buttons
+        */
+        if(!mDriverStartedCommands && isDriverAllowingTeleopCommands()) {
+            mLog.warn("Requesting command start");
+            mDriverStartedCommands = true;
+            updateVisionCommands();
+        /*
+        If the driver started the commands that the superstructure is running and then released the button,
+        stop running commands.
+        */
+        } else if(mSuperstructure.isRunningCommands() && mDriverStartedCommands && !isDriverAllowingTeleopCommands()) {
+            mLog.warn("Requesting command stop: driver no longer allowing commands");
+            mDriverStartedCommands = false;
+            mSuperstructure.stopRunningCommands();
+        } else if(mSuperstructure.isRunningCommands() && isAutoOverridePressed()) {
+            mLog.warn("Requesting command stop: override pressed");
+            mSuperstructure.stopRunningCommands();
         }
-        updateCommands();
 
-    }
-
-    private void updateCommands() {
+        // Teleop control
+        if (!mSuperstructure.isRunningCommands()) {
+            updateDriveTrain();
+        } 
 
-        runCommandQueue = false;
-        for(ELogitech310 l : SystemSettings.kTeleopCommandTriggers) {
-            if(mDriverInputCodex.isSet(l)) {
-                runCommandQueue = true;
-            }
-        }
+    }
 
-        if (shouldInitializeCommandQueue()) {
-            desiredCommandQueue.clear();
-            ETrackingType trackingType = null;
-            // Switch the limelight to a pipeline and track
-            if(mDriverInputCodex.isSet(DriveTeamInputMap.DRIVER_TRACK_TARGET_BTN)) {
-                trackingType = ETrackingType.TARGET_LEFT;
-            } else if(mDriverInputCodex.isSet(DriveTeamInputMap.DRIVER_TRACK_CARGO_BTN)) {
-                trackingType = ETrackingType.CARGO_LEFT;
-            } else if(mDriverInputCodex.isSet(DriveTeamInputMap.DRIVER_TRACK_HATCH_BTN)) {
-                trackingType = ETrackingType.HATCH_LEFT;
+    private void updateDriveTrain() {
+        if(mData.driverinput.isSet(DriveTeamInputMap.DRIVER_THROTTLE_AXIS) &&
+           mData.driverinput.isSet(DriveTeamInputMap.DRIVER_TURN_AXIS) &&
+           mData.driverinput.isSet(DriveTeamInputMap.DRIVER_SUB_WARP_AXIS)) {
+            double rotate = mData.driverinput.get(DriveTeamInputMap.DRIVER_TURN_AXIS);
+            double throttle = -mData.driverinput.get(DriveTeamInputMap.DRIVER_THROTTLE_AXIS);
+
+//		    throttle = EInputScale.EXPONENTIAL.map(throttle, 2);
+            rotate = EInputScale.EXPONENTIAL.map(rotate, 2);
+            rotate = Util.limit(rotate, 0.7);
+
+            if (mData.driverinput.get(DriveTeamInputMap.DRIVER_SUB_WARP_AXIS) > DRIVER_SUB_WARP_AXIS_THRESHOLD) {
+                throttle *= SystemSettings.kSnailModePercentThrottleReduction;
```

Comment:
Will throttle ever get too big or too small with this logic?

Reply:
Technically, throttle can be oversaturated.  Oversaturation is handled in the SRX if we don't handle it ourselves.

The throttle reduction constant reduces power.  Oversaturation isn't an issue here.